models/recently_pushed.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package models
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"code.gitea.io/gitea/models/activities"
+	"code.gitea.io/gitea/models/db"
+	"code.gitea.io/gitea/models/repo"
+	"code.gitea.io/gitea/models/user"
+
+	"xorm.io/builder"
+)
+
+type RecentlyPushedBranches struct {
+	Repo       *repo.Repository
+	BaseRepo   *repo.Repository
+	BranchName string
+	Time       time.Time
+}
+
+// GetRecentlyPushedBranches returns all actions where a user recently pushed but no PRs are created yet.
+func GetRecentlyPushedBranches(ctx context.Context, u *user.User) (recentlyPushedBranches []*RecentlyPushedBranches, err error) {
+	limit := time.Now().Add(-24 * time.Hour).Unix()
```

Comment:
24 hours seems a bit long. I have a feeling this is much shorter on GitHub. Anyone knows how long it is? I would say 2-4 hours max.

Reply:
As per https://stackoverflow.com/a/23129795/808699 it seems to be 2 hours.

Reply:
I've made it two hours now.

If there are people who want to change this in the future, we may add a config variable to do this.